tracer/test/Datadog.Trace.Tests/Configuration/TracerSettingsTests.cs

```diff
@@ -775,9 +835,18 @@ public void IsActivityListenerEnabled(string value, string fallbackValue, string
             const string otelKey = ConfigurationKeys.OpenTelemetry.SdkDisabled;
 
             var source = CreateConfigurationSource((ConfigurationKeys.FeatureFlags.OpenTelemetryEnabled, value), (fallbackKey, fallbackValue), (otelKey, otelValue));
-            var settings = new TracerSettings(source);
+            var errorLog = new OverrideErrorLog();
+            var settings = new TracerSettings(source, NullConfigurationTelemetry.Instance, errorLog);
 
             settings.IsActivityListenerEnabled.Should().Be(expected);
+            Count? metric = (value ?? fallbackValue, otelValue) switch
+            {
+                (null, "uses_default_value") => Count.OpenTelemetryConfigInvalid,
+                (not null, not null) => Count.OpenTelemetryConfigHiddenByDatadogConfig,
+                _ => null,
+            };
```

Comment:
Can we do the following instead so it reads more like the property-based logic that's done by the library?

```suggestion
            Count? metric = (value ?? fallbackValue, otelValue?.ToLower()) switch
            {
                (null, "true") => null,
                (null, _) => Count.OpenTelemetryConfigInvalid,
                (not null, not null) => Count.OpenTelemetryConfigHiddenByDatadogConfig,
                _ => null,
            };
```

Reply:
I guess it doesn't matter too much since this switch will show which specific test cases pass/fail

Reply:
Happy to change it 🙂  - I'll do it when I do the batch of updates